include/extractor/edge_based_node.hpp

```diff
@@ -15,43 +15,48 @@ namespace osrm
 namespace extractor
 {
 
+struct RoadSegment
```

Comment:
@TheMarex yeah, good idea.  The first round here was to see just how much space saving there was to be had.

In theory we can cram +/- 1m precision lon/lat values into 22-23 bits each - we may be able to pack lon/lat/fwd_segment_position into 64 bits and only cost an additional 4 bytes.